save-preprocessor/src/main/kotlin/com/saveourtool/save/preprocessor/service/TestDiscoveringService.kt

```diff
@@ -52,18 +56,22 @@ class TestDiscoveringService(
         log.info { "Starting to save new test suites for root test config in $repositoryPath" }
         return Mono.just(repositoryPath)
             .map { it.resolve(testRootPath) }
-            .map { getRootTestConfig(it.absolutePathString()) }
+            .flatMap { blockingToMono { getRootTestConfig(it.absolutePathString()) } }
             .zipWhen { rootTestConfig ->
-                log.info { "Starting to discover test suites for root test config ${rootTestConfig.location}" }
-                discoverAllTestSuites(
-                    rootTestConfig,
-                    sourceSnapshot,
-                ).toMono()
+                {
+                    log.info { "Starting to discover test suites for root test config ${rootTestConfig.location}" }
+                    discoverAllTestSuites(
+                        rootTestConfig,
+                        sourceSnapshot,
+                    )
+                }.toMono()
```

Comment:
I think here should be `blockingMono` since `discoverAllTestSuites` operates with FS

Reply:
As you can see, `discoverAllTestSuites()` is annotated with `@NonBlocking`.

Indeed, if I'm not mistaken, it only processes TOML content once it's already read into memory.

Reply:
There're three `@Blocking` operations, however: getRootTestConfig(), `getAllTests()`, and `discoverAllTestsIntoMap()`.